src/Http/ServerRequest.php

```diff
@@ -1389,14 +1389,27 @@ protected function _parseAcceptWithQualifier($header)
      *
      * @param string|null $name Query string variable name or null to read all.
      * @return string|array|null The value being read
+     * @deprecated 3.4.0 Use getQuery() and withQueryParams() instead.
      */
     public function query($name = null)
     {
         if ($name === null) {
             return $this->query;
         }
 
-        return Hash::get($this->query, $name);
+        return $this->getQuery($name);
+    }
+
+    /**
+     * Read a specific query value or dotted path.
+     *
+     * @param string $name The name or dotted path to the query param.
+     * @param mixed $default The default value if the named parameter is not set.
+     * @return mixed Query data.
```

Comment:
Technically, yes. But we still always just point out the ones "expected" to be allowed.
Anything else might work, but does not necessarily mean it is supported or should be working.
Also, we also try to not used mixed and be more concrete as mixed does really mean anything and does not help then at all.